x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -349,17 +350,39 @@ private <T extends TransportResponse> void sendWithRemoteAccessHeaders(
 
                 final ThreadContext threadContext = securityContext.getThreadContext();
                 final var contextRestoreHandler = new ContextRestoreResponseHandler<>(threadContext.newRestorableContext(true), handler);
-                authzService.retrieveRemoteAccessRoleDescriptorsIntersection(
-                    remoteClusterAlias,
-                    authentication.getEffectiveSubject(),
-                    ActionListener.wrap(roleDescriptorsIntersection -> {
-                        try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
-                            remoteAccessCredentials.writeToContext(threadContext);
-                            new RemoteAccessAuthentication(authentication, roleDescriptorsIntersection).writeToContext(threadContext);
-                            sender.sendRequest(connection, action, request, options, contextRestoreHandler);
-                        }
-                    }, e -> contextRestoreHandler.handleException(new SendRequestTransportException(connection.getNode(), action, e)))
-                );
+                if (User.isInternal(authentication.getEffectiveSubject().getUser())) {
+                    assertInternalUserExpectedInCrossClusterRequest(authentication.getEffectiveSubject().getUser());
```

Comment:
I have two suggestions for alternatives for your consideration:
1. Move this logic of using empty roleDescriptors into `AuthorizationService` so that this method here does not have to change. I can see an argument for keeping ther interalUser check closer to where request is sent. But any request sent to remote cluster needs to retrieve remote privileges. So handling it in `AuthorizationService#retrieveRemoteAccessRoleDescriptorsIntersection` can actually help it fail earlier on. It is deeper in the callstack which means another caller (if any in future) does not have to handle the special logic on its own. Concretely this means, returning empty for `_system` and throw for all other internal users inside `AuthorizationService#retrieveRemoteAccessRoleDescriptorsIntersection`.
2. Handle `_system` more explicitly with something like `if (SystemUser.is()) { ... } else if (Users.isInternal) { ... } else { ... }`. This is mainly to remove `assertInternalUserExpectedInCrossClusterRequest` which I found a bit implicit.

I slightly prefer 1.

Reply:
We discussed on Slack: going with option 2 since the semantics of `retrieveRemoteAccessRoleDescriptorsIntersection` can be argued to be clearer if the method simply doesn't handle internal users and because special-casing the system user is slightly more consistent with how the system user is handled elsewhere (since the system user does not an actual role, it also makes sense that it's not supported to retrieve any kind of role descriptor for it).